### PR TITLE
Nanite Protocols

### PR DESCRIPTION
**Pull request by UnknownOverlord**
## About The Pull Request
Added Nanites Protocols from TG to add some more late game Nanite Stuff.
2 Types of protocols, Storage and Replication.

Sources ([54710](https://github.com/tgstation/tgstation/pull/54710/files),[48882](https://github.com/tgstation/tgstation/pull/48882))

## Why It's Good For The Game

Additional content always good. 

Can help either accelerate the process of giving people nanites later on or helps improve nanites later on.

## Changelog
:cl:UnknownOverlord, XDTM
add: Added Nanite Protocols
code: edited nanite_designs, nanite_programs, nanites, all_nodes
/:cl:

### CODE REVIEW

**Comment by ZombiesVsPlants** [comment]
The warnings you get from S.L.O.P are pure nightmare fuel
10/10

**Comment by MNarath1** [comment]
@UnknownOverlord why does this remove CaveGenerator.dm????
out of the dme file

File: beestation.dme [hunk 1]
```diff
@@ -574,9 +574,9 @@
 #include "code\datums\looping_sounds\machinery_sounds.dm"
 #include "code\datums\looping_sounds\weather.dm"
 #include "code\datums\mapgen\_MapGenerator.dm"
+#include "code\datums\mapgen\CaveGenerator.dm"
 #include "code\datums\mapgen\JungleGenerator.dm"
 #include "code\datums\mapgen\biomes\_biome.dm"
-#include "code\datums\mapgen\CaveGenerator.dm"
```
Comment (MNarath1) [inline beestation.dme]: Hmm wait this doesn't really remove but move it still there is no reason to do that at all so just drop that change in the dme file and only keep the new added nanite protocol path i guess this probably happened cause of some problems with vsc when adding the new relative path

**Review by MNarath1 — CHANGES_REQUESTED** [review]

File: beestation.dme (hunk 1 above)
Comment (MNarath1) [reply beestation.dme]: So just remove 577 and put in the path back at 579

**Comment by MNarath1** [comment]
@UnknownOverlord Also add the PR you are porting this from into the description of your PR just copy and past the link in there to properly link to the original source.
Ideally you would also include the  original author as a coauthor too in the commits and in the CL in the CL you can do that by writing the original author name and your own name spaced with a , in the same line as the first :CL: in your discription you can see this only in plaintext view of course.

**Comment by MNarath1** [comment]
For anything else repro conduct etc. i would recommend you read https://github.com/BeeStation/BeeStation-Hornet/blob/master/.github/CONTRIBUTING.md

File: beestation.dme (hunk 1 above)
Comment (UnknownOverlord) [reply beestation.dme]: Thanks for that didn't notice my mistake, Think I Did everything you suggested besides adding his names to commits author (Don't know how to change the author of an already done commit. I did however add it to the changelog i believe)

**Comment by MNarath1** [comment]
Nice one thing trough the author names should be after the :cl:

**Comment by UnknownOverlord** [comment]
Ah sorry moved it behind CL is that good?

**Comment by MNarath1** [comment]
yes should be fine like that

**Comment by MNarath1** [comment]
Also i can understand why you didn't add the coauthor to commits too cause doing that can be a bit exausting sometimes as you need to find out their commit email etc.

**Comment by Crossedfall** [comment]
> Also i can understand why you didn't add the coauthor to commits too cause doing that can be a bit exausting sometimes as you need to find out their commit email etc.

You just have to add `.patch` to the end of the commit.... Example from my most recent merge commit: https://github.com/BeeStation/BeeStation-Hornet/commit/bb616fbe8d310a2a2ad3b79355cdc73c966c6113.patch

**Comment by UnknownOverlord** [comment]
Ah sorry didn't know I had to do that my mistake, believe I've done that now because it auto-checked run tests

**Comment by MNarath1** [comment]
speaking about patch i don't know if you intended to commit with the email you did if not i suggest you look up a guide how to change the commit email (possible to your placeholder email) of already commited changes and then force push them to update the PR.

**Comment by UnknownOverlord** [comment]
I'm just gonna close the PR cause I feel like messed up everything plus I made it into the master branch instead of making a special one. So i will close it and open up another late most likely possibly with more stuff.

**Pull request closed without merging**